presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java

```diff
@@ -388,12 +389,23 @@ private static void initializeDeserializer(Configuration configuration, Deserial
         try {
             configuration = copy(configuration); // Some SerDes (e.g. Avro) modify passed configuration
             deserializer.initialize(configuration, schema);
+            getConfigurationErrors(deserializer)
+                    .ifPresent(configurationErrors -> { throw new RuntimeException("There are configuration errors: " + configurationErrors); });
         }
         catch (SerDeException | RuntimeException e) {
             throw new RuntimeException("error initializing deserializer: " + deserializer.getClass().getName(), e);
         }
     }
 
+    @SuppressWarnings("deprecation")
+    private static Optional<String> getConfigurationErrors(Deserializer deserializer)
+    {
+        if (deserializer instanceof AbstractSerDe) {
+            return Optional.ofNullable(((AbstractSerDe) deserializer).getConfigurationErrors());
```

Comment:
Why not just throw the runtime exception from this method?  Something like:
```java
private static void getValidateDeserializer(Deserializer deserializer) {
    if (deserializer instanceof AbstractSerDe) {
        SomeType configurationErrors = ((AbstractSerDe) deserializer).getConfigurationErrors();
        if (configurationErrors != null) {
            throw new RuntimeException("There are configuration errors: " + configurationErrors);
        }
    }
}
```

Reply:
Done